src/enclave_proc/resource_manager.rs

```diff
@@ -1007,13 +1035,26 @@ fn write_eif_to_regions(
     regions: &[MemoryRegion],
     image_write_offset: usize,
 ) -> NitroCliResult<()> {
+    let link_path = read_link(PathBuf::from(format!(
+        "/proc/self/fd/{}",
+        eif_file.as_raw_fd()
+    )));
+    let eif_path = match link_path {
+        Ok(link_path) => String::from(
+            link_path
+                .to_str()
+                .unwrap_or("Invalid unicode EIF file name"),
+        ),
+        Err(_) => String::from("Unknown file"),
+    };
     let file_size = eif_file
         .metadata()
         .map_err(|_| {
             new_nitro_cli_failure!(
                 "Failed to obtain EIF file metadata",
                 NitroCliErrorEnum::FileOperationFailure
             )
+            .add_info(vec![eif_path.as_str(), "Get metadata"])
```

Comment:
As far as I can tell, metadata will only fail here if the fd is bad, which is also not really related to the file itself, so I don't think we have to print out the file path here.

Reply:
Removed it in the latest revision.